src/gtest/test_merkletree.cpp

```diff
@@ -235,6 +238,29 @@ TEST(merkletree, vectors) {
     test_tree<ZCTestingIncrementalMerkleTree, ZCTestingIncrementalWitness>(root_tests, ser_tests, witness_ser_tests, path_tests);
 }
 
+TEST(merkletree, emptyroots) {
+    Array empty_roots = read_json(std::string(json_tests::merkle_roots_empty, json_tests::merkle_roots_empty + sizeof(json_tests::merkle_roots_empty)));
+    Array::iterator root_iterator = empty_roots.begin();
+
+    libzcash::EmptyMerkleRoots<64, libzcash::SHA256Compress> emptyroots;
+
+    for (size_t depth = 0; depth <= 64; depth++) {
+        expect_test_vector(root_iterator, emptyroots.empty_root(depth));
+    }
+
+    // Double check that we're testing (at least) all the empty roots we'll use.
+    ASSERT_TRUE(INCREMENTAL_MERKLE_TREE_DEPTH <= 64);
```

Comment:
This is an excellent check to do. (It would be nice if it were a static check that failed during compilation, but a unit test will do.)